### PR TITLE
Email templates

### PR DESCRIPTION
**Pull request by mbertrand**
#### What are the relevant tickets?
Closes https://trello.com/c/ch9rjBlN/28-update-notification-email-when-video-is-ready

#### What's this PR do?
- Adds HTML email templates to OVS
- Does not yet remove the `NotificationEmail` model - my thought is that it should be removed in a subsequent PR in case this one needs to be reverted.

#### How should this be manually tested?
- Set `MAILGUN_RECIPIENT_OVERRIDE=<your_email>` and other MAILGUN .env variables (you can get them from CI)
- Set `ODL_VIDEO_FROM_EMAIL=MIT ODL Video <odl-video-support@mit.edu>`
- Set `ODL_VIDEO_SUPPORT_EMAIL=MIT ODL Video <odl-video-support@mit.edu>`
- Run the following, you should get 3 emails afterward:
```
from mail.tasks import *
from ui.models import *
video = Video.objects.order_by("-created_at").first()
for status in [VideoStatus.COMPLETE, VideoStatus.TRANSCODE_FAILED_VIDEO, VideoStatus.UPLOAD_FAILED]:
    video.status = status
    send_notification_email(video)
```

#### Screenshots (if appropriate)

SUCCESS:

Screen Shot 2020-04-23 at 1 43 21 PM

ERROR LIKELY DUE TO INVALID FORMAT:

Screen Shot 2020-04-23 at 1 43 36 PM

OTHER ERRORS:

Screen Shot 2020-04-23 at 1 24 09 PM

### CODE REVIEW

**Comment by mbertrand** [comment]
@pdpinch @abdulkdawson screenshots attached (taken from Outlook)

File: mail/templates/success/body.html [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{% extends "email_base.html" %}
+{% load static %}
+
+{% block content %}
+  <tr>
+      <td style="background-color: #ffffff;">
+        <p>Hello, your video is ready!</p>
+
+        <p>Title: <a href={{ video_url }}>{{video_title}}</a></p>
+        <p>Collection Title: <a href={{ collection_url }}>{{collection_title}}</a></p>
```
Comment (pdpinch) [inline mail/templates/success/body.html:10]:
```suggestion
        <p>Video: <a href={{ video_url }}>{{video_title}}</a></p>
        <p>Collection: <a href={{ collection_url }}>{{collection_title}}</a></p>
```

I'm confused by the labels "Title" and "Collection Title" I think "Video" and "Collection" will work better.

File: mail/templates/success/body.html [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{% extends "email_base.html" %}
+{% load static %}
+
+{% block content %}
+  <tr>
+      <td style="background-color: #ffffff;">
+        <p>Hello, your video is ready!</p>
```
Comment (pdpinch) [inline mail/templates/success/body.html:7]:
```suggestion
        <p>Hello, your video is now available on the OVS web site.</p>
```
I think we can collapse the two sentences into one. They say the same thing.

File: mail/templates/success/body.html [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+        <p>Title: <a href={{ video_url }}>{{video_title}}</a></p>
+        <p>Collection Title: <a href={{ collection_url }}>{{collection_title}}</a></p>
+        <p>Has finished processing and is now available on the OVS web site</p>
```
Comment (pdpinch) [inline mail/templates/success/body.html:11]:
```suggestion
```

I suggest moving this line up above.

Comment (mbertrand) [reply mail/templates/success/body.html]: @pdpinch I made those changes and updated the screenshots

**Comment by noisecapella** [comment]
I don't think the material-icons will work inside email templates. You might need to embed images instead.
Screenshot from 2020-04-16 17-28-37

File: mail/constants.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+""" Constants for mail """
+from ui.constants import VideoStatus
+
+EMAIL_SUCCESS = "success"
+EMAIL_INVALID_FORMAT = "invalid_format_error"
+EMAIL_OTHER_ERROR = "other_error"
+
+STATUS_TO_NOTIFICATION = {
+    VideoStatus.COMPLETE: EMAIL_SUCCESS,
+    VideoStatus.TRANSCODE_FAILED_INTERNAL: EMAIL_OTHER_ERROR,
```
[the remark this answers is not in the document]
Comment (mbertrand) [reply mail/constants.py:10]: Before, they were paired to possible values of the `NotificationEmail.notification_type` attribute.  Now they need to pair up with the names of the templates.

**Comment by mbertrand** [comment]
@noisecapella which mail program are you using?  The icon was showing up for me in outlook after I chose to download images.  The base template includes `<link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons" />`

**Comment by mbertrand** [comment]
Regarding above, @pdpinch do you want the user to have to choose to download images to see the film icon, or should it just be removed?

**Comment by mbertrand** [comment]
@pdpinch @noisecapella the film icon seems to work in outlook but not gmail.  Converting to an image instead of material icon css would still require users to choose to download images to see it.  Should I just get rid of it altogether?

**Comment by pdpinch** [comment]
Yeah, let's get rid of it. I don't think it's worth complicating a notification email.

**Comment by mbertrand** [comment]
The icon has been removed (and the screenshot has been updated)

**Comment by noisecapella** [comment]
gmail looks a little different from the screenshot:
Screenshot from 2020-04-22 17-14-34

File: mail/templates/invalid_format_error/body.html [hunk 5]
```diff
@@ -0,0 +1,13 @@
+{% extends "email_base.html" %}
+{% block content %}
+  <tr>
+      <td style="background-color: #ffffff;">
+        <p>We’re sorry, there was a problem processing your video</p>
+        <p>Video: <a href={{ video_url }}>{{video_title}}</a></p>
```
Comment (noisecapella) [inline mail/templates/invalid_format_error/body.html:6]: Can we put quote marks around the `{{  }}` here and the other `href` attributes in this PR?

File: mail/templates/success/body.html (hunk 3 above)
Comment (mbertrand) [reply mail/templates/success/body.html]: @pdpinch do you want the signature on the error messages too or just the success email?

Comment (pdpinch) [reply mail/templates/success/body.html]: It should be everywhere, but I'm happy with whatever is easiest.

**Comment by mbertrand** [comment]
Screenshots updated

**Comment by mbertrand** [comment]
@noisecapella ready for another look

**Comment by pdpinch** [comment]
@mbertrand @noisecapella is this ready to merge?

**Comment by noisecapella** [comment]
Sorry, looking at it now

**Pull request merged**